### ISSUE TITLE
AnyVal classes seem to give unused parameter false positives

### ISSUE THREAD

**Issue by johnynek** [HairyFotr/linter#38]
```
src/scala/com/stripe/streamish/Streamish.scala:33: warning: Parameter $this is not used in method copy$extension.
case class InvariantEvents[T](evs: Events[T]) extends AnyVal {
           ^
src/scala/com/stripe/streamish/Streamish.scala:33: warning: Parameter $this is not used in method productPrefix$extension.
case class InvariantEvents[T](evs: Events[T]) extends AnyVal {
           ^
src/scala/com/stripe/streamish/Streamish.scala:33: warning: Parameter $this is not used in method productArity$extension.
case class InvariantEvents[T](evs: Events[T]) extends AnyVal {
           ^
src/scala/com/stripe/streamish/Streamish.scala:33: warning: Parameter $this is not used in method canEqual$extension.
case class InvariantEvents[T](evs: Events[T]) extends AnyVal {
           ^
warning: there were three feature warnings; re-run with -feature for details
5 warnings found
```
those methods are all compiler generated when using `AnyVal`. `-P:linter:disable:UnusedParameter` fixes it, but that is a pretty heavy hammer to drop on default settings.

**Comment by HairyFotr** [comment]
Thanks for opening the issue. Synthetic methods and parameters weren't ignored - it should be fixed now in 0.1-SNAPSHOT.

**Comment by HairyFotr** [comment]
Did a bit more testing and released 0.1.14.
